rotkehlchen/tasks/calendar.py

```diff
@@ -47,3 +70,98 @@ def delete_past_calendar_entries(database: DBHandler) -> None:
             'DELETE FROM calendar WHERE timestamp < ? AND auto_delete=1',
             (ts_now(),),
         )
+
+
+def maybe_create_ens_reminders(database: DBHandler) -> None:
+    """Check ENS registration and renewal history events and create reminders if needed."""
+    db_calendar = DBCalendar(database=database)
+    db_history_events = DBHistoryEvents(database=database)
+    customizable_date = CustomizableDateMixin(database=database)
+    with database.conn.read_ctx() as cursor:
+        registrations: list['EvmEvent'] = db_history_events.get_history_events(
+            cursor=cursor,
+            has_premium=True,  # to not hit the free limit while internal processing
+            group_by_event_ids=False,
+            filter_query=EvmEventFilterQuery.make(
+                and_op=True,
+                counterparties=[CPT_ENS],
+                event_types=[HistoryEventType.TRADE],
+                event_subtypes=[HistoryEventSubType.SPEND],
+            ),
+        )
+        renewals: list['EvmEvent'] = db_history_events.get_history_events(
+            cursor=cursor,
+            has_premium=True,  # to not hit the free limit while internal processing
+            group_by_event_ids=False,
+            filter_query=EvmEventFilterQuery.make(
+                and_op=True,
+                counterparties=[CPT_ENS],
+                event_types=[HistoryEventType.RENEW],
+                event_subtypes=[HistoryEventSubType.NONE],
+            ),
+        )
+
+    with database.conn.read_ctx() as cursor:
+        blockchain_accounts = database.get_blockchain_accounts(cursor=cursor)
+
+    ens_to_event: dict[str, tuple[int, EvmEvent]] = {}
+    for ens_event in registrations + renewals:
+        if (
+            ens_event.location_label is not None and
+            ens_event.extra_data is not None and
+            'fullname' in ens_event.extra_data and
+            'expires' in ens_event.extra_data and
+            ens_event.extra_data['expires'] > ts_now()  # those with expiry in the future
+        ):
+            ens_name: str = ens_event.extra_data['fullname']
+            ens_expires: Timestamp = ens_event.extra_data['expires']
+            blockchain = ChainID.deserialize(ens_event.location.to_chain_id()).to_blockchain()
+            if (calendar_entries := db_calendar.query_calendar_entry(
+                filter_query=CalendarFilterQuery.make(
+                    and_op=True,
+                    name=ens_name,
+                    addresses=[OptionalBlockchainAddress(
```

Comment:
an ethereum address can have multiple ens names. You need to check the timestamp too

Reply:
using the name filter as `f'{ens_name} expiry'` so that should be enough to differentiate them